AsyncDisplayKit/_ASTransitionContext.h

```diff
@@ -0,0 +1,30 @@
+//
+//  _ASTransitionContext.h
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 2/4/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "ASContextTransitioning.h"
+
+@class ASLayout;
+@class _ASTransitionContext;
+
+@protocol _ASTransitionContextDelegate <NSObject>
+
+- (void)transitionContext:(_ASTransitionContext *)context didComplete:(BOOL)didComplete;
+- (CGRect)transitionContext:(_ASTransitionContext *)context initialFrameForNode:(ASDisplayNode *)node;
+- (CGRect)transitionContext:(_ASTransitionContext *)context finalFrameForNode:(ASDisplayNode *)node;
+
+@end
+
+@interface _ASTransitionContext : NSObject <ASContextTransitioning>
```

Comment:
This file seems like it provides an interesting abstraction, but the only use case I could see, it seemed like we might just be able to call self directly to replace it. However, it feels like you have a plan for this that is not fully built out yet… Is that support in ASViewController?